### ISSUE TITLE
LetsEncrypt Does Not Renew Successfully

### ISSUE THREAD

**Issue by stone212** [nextcloud-snap/nextcloud-snap#750]
I am handling SSL with `# nextcloud.enable-https`. My certificate expired today and my browser gave me an error. I re-ran `nextcloud.enable-https` and the wizard gave no errors.  But the certificate is not being read.  Is anyone else having this problem?  I am trying to find the source but I am not familiar yet with how the snap architecture works.

**Comment by stone212** [comment]
Update:

I *think* there are two problems:

1. For some reason the certificate did not update itself correctly.  I have not looked into this.

2. Because there is no way to delete a certificate with `nextcloud.enable-https` as with `certbot delete` (see https://github.com/nextcloud/nextcloud-snap/issues/690), when I manually updated the certificate it created a second directory for the new certificates (because the old one was not deleted).  And the snap's config was looking for the first/default directory.  

The solution was to delete the old directory and rename the new one to defaultdomainname.com

**Comment by xkl** [comment]
once I had a similar issue. The cause was a non-standard port for https.

**Comment by kyrofa** [comment]
You might be able to see what the issue was from the journal of the renew-certs job:

`journalctl -n 100 -u snap.nextcloud.renew-certs` might do it, you can adjust the `-n` param, too.

**Comment by stone212** [comment]
@kyrofa Thanks for that.  I don't know where I would find docs on `snap.nextcloud.renew-certs` so I would not have guessed how to find this without your help.  Here is the part that I think matters:

```
Oct 19 10:50:07 files nextcloud.renew-certs[3845]:   File "/snap/nextcloud/8971/lib/python2.7/site-packages/certbot/storage.py", line 498, in _check_symlinks
Oct 19 10:50:07 files nextcloud.renew-certs[3845]:     "expected {0} to be a symlink".format(link))
Oct 19 10:50:07 files nextcloud.renew-certs[3845]: CertStorageError: expected /var/snap/nextcloud/current/certs/certbot/config/live/files.ourdomain.com-0001/cert.pem to
Oct 19 10:50:07 files nextcloud.renew-certs[3845]: Renewal configuration file /var/snap/nextcloud/current/certs/certbot/config/renewal/files.ourdomain.com-0001.conf is 
Oct 19 10:50:07 files nextcloud.renew-certs[3845]: - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
```

I think this is the important part because it co-relates to Item 2. that I wrote above which was:

```
Because there is no way to delete a certificate with nextcloud.enable-https as with certbot delete (see #690), when I manually updated the certificate it created a second directory for the new certificates (because the old one was not deleted). And the snap's config was looking for the first/default directory.
```

So a small bug.  Not critical.  The workaround I used worked perfectly well.

**Comment by kyrofa** [comment]
That's annoying, I apologize. Would you say the root cause of this is #690, then? i.e. if we fixed #690, is that also the fix for this? If so, we can probably close this one and track that one.

**Comment by stone212** [comment]
@kyrofa Actually no.  Fixing #690 would make this error go away yes but that would not be because this error is fixed.  It would be because it is never exposed.  The problem (I think) is a combination of:

1. Unable to delete old certificate
2. the automatic letsencrypt renewal does not overwrite symlinks correctly in the case that an old certificate isn't deleted
